transformer_engine/pytorch/csrc/extensions/pybind.cpp

```diff
@@ -200,8 +201,8 @@ PYBIND11_MODULE(TORCH_EXTENSION_NAME, m) {
         py::arg("otype"), py::arg("sm_margin"), py::arg("zero_centered_gamma"));
   m.def("rmsnorm_bwd", &transformer_engine::pytorch::rmsnorm_bwd, "Backward of RMSNorm");
   m.def("fused_multi_quantize", &transformer_engine::pytorch::fused_multi_quantize,
-        "Fused Multi-tensor Cast + Transpose", py::arg("input_list"), py::arg("output_list"),
-        py::arg("quantizer_list"), py::arg("otype"));
+        "Fused Multi-tensor Cast + Transpose", py::arg("input_list"), py::arg("input_view"),
+        py::arg("m_splits"), py::arg("quantizer_list"), py::arg("otype"));
```

Comment:
I think the function name should be a bit more descriptive since it's now more MoE-specific than before:
```suggestion
  m.def("fused_multi_quantize_from_view", &transformer_engine::pytorch::fused_multi_quantize,
        "Fused split + multi-tensor quantize", py::arg("input_list"), py::arg("input_view"),
        py::arg("m_splits"), py::arg("quantizer_list"), py::arg("otype"));
```

Reply:
Yeah, I also plan to remove having to pass in the input_list because torch.split is another big source of cpu overhead. 

However, there are two cases to consider for removing torch split: 
- FP8: split happens in multi-quantize
- BF16: split happens right before gemm

It would make the code cleaner if in the bf16 path, there will be a line that explicitly says it splits the views, but since we are in the MoE path, passing N py objects into C++ and fetch it back is too slow. Still I am not sure how to avoid that torch.split though.

Reply:
If we go that way, the FP8 operation should have a name like `fused_split_quantize`. Then the following code seems logical:
```python
if with_quantized_compute:
    inputmats = tex.fused_split_quantize(inp_view, m_splits, input_quantizers)
else:
    inputmats = torch.split(inp_view, m_splits)

general_grouped_gemm(weights_fp8, inputmats, ...)
```
If we really wanted to optimize the BF16 path, we could modify the grouped GEMM function to take in the full tensor and split it internally, but I don't like how that would result in different APIs for BF16 and FP8.